front/js/category.es6

```diff
@@ -33,6 +42,70 @@
     $(DOM.$productsTable).on('click', buyProduct);
     DOM.$showMoreLink.click(loadProducts);
     DOM.$searchFilter.keyup(helpers.debounce(filterProducts, 400));
+    DOM.$filtersApplyBtn.click(loadFilteredProducts);
+    DOM.$filtersWrapper.on('click', 'input', toggleApplyBtnState);
+  }
+
+  const TAGS_TYPE_DELIMITER = '-or-';
+  const TAGS_GROUP_DELIMITER = '-and-';
+
+  function serializeTags(tags) {
+    const tagsByGroups = tags.reduce((group, item) => {
+      const groupId = item.group;
+      group[groupId] = group[groupId] || [];  // Ignore ESLintBear (no-param-reassign)
+      group[groupId].push(item.slug);
+      return group;
+    }, {});
+
+    return Object.keys(tagsByGroups).reduce((previous, current) => {
+      const delim = previous ? TAGS_GROUP_DELIMITER : '';
+      return previous + delim + tagsByGroups[current].join(TAGS_TYPE_DELIMITER);
+    }, '');
+  }
+
+  function parseTags(string) {
+    return [].concat(...(
```

Comment:
We can use `.reduce()` instead of `[].concat(...())`.  It seems it looks more readable

```js
string
  .split(TAGS_GROUP_DELIMITER)
  .map(group => group.split(TAGS_TYPE_DELIMITER))
  .reduce((acc, e) => acc.concat(e), [])
```